Week 1 : 683380435-0
นายภาสุ สมมีย์




Lab 1.1

In [ ]:
print("Hello, World!")

Hello, Hello!


Lab 1.2

2. ประกาศตัวแปรและแสดงผล:


In [ ]:
age = 20
gpa = 3.99
color = "Purple"
student = True

print("Age = ",age ,type(age))
print("GPA = ",gpa,type(gpa))
print("Favorite_Color = ",color,type(color))
print("Student = ",student,type(student))

Age =  210 <class 'int'>
GPA =  3.99 <class 'float'>
Favorite_Color =  Purple <class 'str'>
Student =  True <class 'bool'>


3. ประกาศตัวแปรและคํานวณ:


In [ ]:
num1 = 1
num2 = 2
sum = num1 + num2
print("Sum = ",sum)

Sum =  3


4. ประกาศตัวแปรStringและแสดงผลด้วย f-String:


In [ ]:
fname = "Phasu"
lname = "Sommee"
flname = fname + " " + lname
print(f"Hello, {flname}")

Hello, Phasu Sommee


5. การรับค่าจากผู้ใช้ (User Input) และการแปลงประเภทข้อมูล:


In [ ]:
num = input("Input num : ")
print("Type A  : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

Input num : 5
Type A  :  <class 'str'>
5  + 5 =  10


Activity 1.1: Class Discussion - Uses of Scripting
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.


คำตอบ: การเขียนสคริปต์ด้วยภาษา Python เป็นเครื่องมือที่มีประสิทธิภาพสูง สามารถใช้ใน การจัดการข้อมูล การจำลองสถานการณ์และคำนวณทางวิทยาศาสตร์ทั่วไปได้

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))


13.5
<class 'float'>


ได้ผลลัพธ์: `13.5`

ทำไมถึงเป็นประเภทนี้ : เมื่อนำ `int`
มาบวกกับ `float` ภาษา Python
แปลงให้เป็น `float` โดยอัตโนมัติ

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: `Alice is 30 years old.`

ทำไมถึงเป็น : เพราะ data type `String` สามารถ รับข้อมูลที่เป็นตัวเลขได้

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: `False`

ทำไมถึงเป็น : นิพจน์ `is_sunny` และ `is_raining` ตรวจสอบว่าเงื่อนไขทั้งสองเป็นจริงหรือไม่ เนื่องจาก `is_sunny` เป็นจริงและ `is_raining เป็นเท็จ ผลลัพธ์จะเป็น `False`

In [ ]:
# Puzzler 4
# What happens here?
num_str = input("Enter a number: ")
result = int(num_str) + 5
print(result)

Enter a number: 5


TypeError: can only concatenate str (not "int") to str

ได้ผลลัพธ์: `Error`

ทำไมถึงเป็น : เพราะ เราไม่สามารถ บวก `String` ลงใน Data Type `int` โดยตรงได้ จึงต้องแปลงค่าที่รับมาเป็น `int` ก่อน

In [ ]:

num_str = input("Enter a number: ")
result = int(num_str) + 5
print(result)

Enter a number: 5
10
